### ISSUE TITLE
python.lang.maintainability.useless-innerfunction.useless-inner-function false positive

### ISSUE THREAD

**Issue by rcoh** [returntocorp/sgrep-rules#563]
**Describe the bug**
With pytest-benchmark, you can define benchmarks like this:

```python
def some_test(benchmark):
  @benchmark
  def func():
     time.sleep(5)
```

This produces: 
```
semgrep python.lang.maintainability.useless-innerfunction.useless-inner-function
⚠ *** finding(s) due to staged changes in 9.82 s
     > semgrep/tests/performance/test_ci_perf.py:48
```

**Expected Result**
I'm not sure the perfect behavior, but probably in general, we shouldn't trigger `useless-inner-function` if the function has a decorator on it.

**Comment by DrewDennison** [comment]
I think useless-innerfunction is just straight up broken cc @ievans

**Comment by minusworld** [comment]
There seems to be an issue matching annotation metavariables: https://semgrep.live/10Qb

**Comment by ievans** [comment]
Lots more issues at https://semgrep.live/scan/535aa92a-b79d-478c-8b82-b31e2310e154
